docs/synapse/userguides/ug018_storm_ref_misc.rst

```diff
@@ -13,7 +13,54 @@ Todo
 
 limit()
 -------
-Todo
+Provides a hard limit on the number of nodes which are emitted by the ``limit()`` operator.
+This also back-propagates a ``limit=`` argument to the previous query operator.
+
+
+Optional parameters:
+
+* **Return limit:** specify the maximum number of nodes returned by the ``pivot()`` query.
+
+  * ``limit=`` (operator syntax)
+
+**Operator syntax:**
+
+.. parsed-literal::
+
+  **limit( *<limit>* )**
+
+**Macro syntax:**
+
+N/A
```

Comment:
FYI only: There is "sort of" a macro syntax for "limit" - the carat ( ^ ) symbol. However, I think it only works / applies when using by handlers, so we can skip the discussion here and talk about it in the by handler docs:

ask inet:fqdn^10*tag=aka.foocorp.foothreat -> retrieve inet:fqdns tagged "aka.foocorp.foothreat" and limit the results to 10

However:

ask inet:fqdn^10 (lift inet:fqdns, limit results to 10) fails.

Reply:
That macro syntax is only relevant to the liift-by macro syntax. The second item sounds like a parser failure somewhere.

Reply:
Actually just tried that second thing...and it worked.

No words.

Reply:
no action